test/storage/http.test.js

```diff
@@ -21,25 +21,99 @@ const fs = require('fs-extra');
     });
     afterEach( () => {
-        http.downloadFile = oldDownloadFileHttpTransfer;
+        // TODO: switch tests that are using this to proxyquire
+        http.downloadFileConcurrently = oldDownloadFileHttpTransfer;
         nock.cleanAll();
+        // mockFs messes with rewire and proxyquire loading in of modules
+        // we must make sure to restore mockFs before using these modules
         mockFs.restore();
         delete process.env.__OW_ACTION_NAME;
         delete process.env.ASSET_COMPUTE_DISABLE_RETRIES;
     });
 
+    describe('getMaxConcurrent', () => {
+        it("no containerMemorySize available, default concurrency",  async () => {
+            const maxConcurrent = await getMaxConcurrent(DEFAULT_PREFERRED_PART_SIZE);
+            assert.strictEqual(maxConcurrent, DEFAULT_MAX_CONCURRENT);
+        });
+        it("containerMemorySize is missing or malformated, default concurrency",  async () => {
+            mockFs({ '/sys/fs/cgroup/memory/memory.limit_in_bytes': 'not a valid number' });
+            const maxConcurrent = await getMaxConcurrent(DEFAULT_PREFERRED_PART_SIZE);
+            assert.strictEqual(maxConcurrent, DEFAULT_MAX_CONCURRENT);
+        });
+        it("containerMemorySize is 40gb, use default concurrency",  async () => {
+            // this is the container memory size for worker-pie-large
+            // although there is a lot of memory available and we could use higher concurrency,
+            // we still use the default concurrency
+            const containerMemorySizeInBytes = '41943040000'; // 1gb
+            mockFs({ '/sys/fs/cgroup/memory/memory.limit_in_bytes': containerMemorySizeInBytes });
+            const maxConcurrent = await getMaxConcurrent(DEFAULT_PREFERRED_PART_SIZE);
+            assert.strictEqual(maxConcurrent, DEFAULT_MAX_CONCURRENT);
+        });
+        it("containerMemorySize is 2gb, use default concurrency",  async () => {
+            // this is the container memory size for a lot of our workers
+            const containerMemorySizeInBytes = '2147483648'; // 1gb
+            mockFs({ '/sys/fs/cgroup/memory/memory.limit_in_bytes': containerMemorySizeInBytes });
+            const maxConcurrent = await getMaxConcurrent(DEFAULT_PREFERRED_PART_SIZE);
+            assert.strictEqual(maxConcurrent, DEFAULT_MAX_CONCURRENT);
+        });
+        it("containerMemorySize is 1024mb, use default concurrency",  async () => {
+            // this is the container memory size for a lot of our workers
+            const containerMemorySizeInBytes = '1073741824'; // 1gb
+            mockFs({ '/sys/fs/cgroup/memory/memory.limit_in_bytes': containerMemorySizeInBytes });
+            const maxConcurrent = await getMaxConcurrent(DEFAULT_PREFERRED_PART_SIZE);
+            assert.strictEqual(maxConcurrent, DEFAULT_MAX_CONCURRENT);
+        });
+        it("containerMemorySize is barely too small, adjust concurrency down",  async () => {
```

Comment:
Sounds clearer to me? 
```suggestion
        it("containerMemorySize is slightly too small, adjust concurrency down",  async () => {
```

Reply:
I made it even more explicit, by saying it is 1 byte too small: `containerMemorySize is 1 byte too small, adjust concurrency down`